library/Icingadb/Web/Control/SearchBar/ObjectSuggestions.php

```diff
@@ -244,70 +250,81 @@ protected function fetchColumnSuggestions($searchTerm)
                 foreach ($this->customVarSources as $relation => $label) {
                     if (isset($customVar->$relation)) {
                         $varRelation = $relation . '.vars.' . $search;
-                        $varLabel = sprintf($label, $name);
-                        if ($name === $exactSearchTerm) {
-                            $exactVarMatches[$varRelation] = $varLabel;
-                        } elseif ($this->matchSuggestion($varRelation, $varLabel, $searchTerm)) {
-                            $varSuggestions[$varRelation] = $varLabel;
+                        if ($titleAdded === false) {
+                            $this->addHtml(HtmlElement::create(
+                                'li',
+                                ['class' => static::SUGGESTION_TITLE_CLASS],
+                                t('Best Suggestions')
+                            ));
+
+                            $titleAdded = true;
                         }
-                    }
-                }
-            }
-        }
-
-        // Adjust the number of columns to be suggested based on custom variable count
-        $varCount = count($exactVarMatches) + count($varSuggestions);
-        $colLimit = $halfLimit;
-        if ($varCount < $halfLimit) {
-            $colLimit = $halfLimit + ($halfLimit - $varCount);
-        }
 
-        // Exact custom variable matches first
-        if (! empty($exactVarMatches)) {
-            $this->addHtml(HtmlElement::create(
-                'li',
-                ['class' => static::SUGGESTION_TITLE_CLASS],
-                t('Best Suggestions')
-            ));
+                        $exactVarMatches[] = $varRelation;
```

Comment:
Can't you register `$search` here to avoid checking this over and over while iterating through the sources below?